src/engine/services/event-system.ts

```diff
@@ -0,0 +1,13 @@
+import GameEvent from '../events/event'
+import SpawnEvent from '../events/spawn-event'
+import Game from '../game'
+
+export default class EventSystem {
+
+  static handleEvent(game: Game, event: GameEvent | undefined): void {
+
+    if ((<any>event).constructor.name == (<any>SpawnEvent).name) {
```

Comment:
For casting, use `as`. I read on the difference one time and apparently there are issues that can arise from this notation but not `as` or something like that.